src/converters.ts

```diff
@@ -211,55 +232,73 @@ export function transformCredentialInput(
   const credentialSubject = { ...input.credentialSubject, ...input.vc?.credentialSubject }
   if (!input.sub) {
```

Comment:
This is not a discussion for this PR, but after talking to @awoie I think these defensive IFs in the VC->JWT transformation are not correct. If a sub is already defined I should either THROW, or forcefully overwrite it with input.credentialSubject?.id, because I never want to produce a JWT where these values differ.

Reply:
right, that is indeed an issue when this edge case occurs. I preferred the non-distructive approach in the original transforms, but it can be switched to the overwrite or the `throw` approach. However, that can be a separate task, not necessarily part of this PR which deals with _delete or not delete_

Reply:
I created #64 to deal with this ambiguity